### ISSUE TITLE
MA0106 fix is not possible for .NET Standard 2.0

### ISSUE THREAD

**Issue by hankovich** [meziantou/Meziantou.Analyzer#568]
### Version of the Meziantou.Analyzer NuGet package

2.0.68

### Rule Identifier

MA0106

### Target Framework

netstandard2.0

### C# Language version

C# 12

### Description

MA0106 asks to use [this overload](https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.addorupdate?view=net-8.0#system-collections-concurrent-concurrentdictionary-2-addorupdate-1(-0-system-func((-0-0-1))-system-func((-0-1-0-1))-0)), but it's not available on .nET Standard 2.0, so there's no way to fix it

**Comment by meziantou** [comment]
The rules:
- MA0105 is reported on netstandard2.0 as this is fixable
- MA0106 is not reported on netstand2.0 as the new overload doesn't exist

````c#
ConcurrentDictionary<int, int> dict = new();
var key = 0;
var value = 0;

dict.GetOrAdd(key, i => key);   // MA0105
dict.GetOrAdd(key, i => i);     // ok
dict.GetOrAdd(key, i => value); // MA0106 => not reported on netstandard2.0
````

You can check the target in the error list

<img width="664" alt="image" src="https://github.com/meziantou/Meziantou.Analyzer/assets/509220/b8090252-ed9f-45cd-9177-d4e5da2f9118">

**Comment by meziantou** [comment]
Note that analyzers run per target, and do not have access to other targets. So, it's not possible (and maybe not desired) to not report on net6 when a project multitarget netstandard2.0. You can disable the rule or use `#if` to fix the rule based on the target.
